Makefile

```diff
@@ -1,58 +1,66 @@
 PIP_HOME = $(shell python3 -c "import site; import os; print(os.path.join(site.USER_BASE, 'bin'))" \
 	|| python -c "import site; import os; print(os.path.join(site.USER_BASE, 'bin'))")
 
+VENV := .venv
+BIN := $(VENV)/bin
+PYTHON := $(BIN)/python
+MANAGE := HKNWEB_MODE='dev' $(PYTHON) ./manage.py
+
 IP ?= 127.0.0.1
 PORT ?= 3000
 
 .PHONY: dev
 dev:
-	HKNWEB_MODE='dev' pipenv run python ./manage.py runserver $(IP):$(PORT)
+	$(MANAGE) runserver $(IP):$(PORT)
 
 .PHONY: livereload
 livereload:
-	HKNWEB_MODE='dev' pipenv run python ./manage.py livereload $(IP):$(PORT)
+	$(MANAGE) livereload $(IP):$(PORT)
 
-setup: pipenv venv migrate
+.PHONY: venv
+venv:
+	python3 -m venv $(VENV)
+	@echo "When developing, activate the virtualenv with 'source .venv/bin/activate' so Python can access the installed dependencies."
 
-.PHONY: pipenv
-pipenv:
-	pip3 install --user pipenv || pip install --user pipenv
-	@echo Please add $(PIP_HOME) to your PATH variable.
+.PHONY: install-prod
+install-prod:
+	# For issues with binary packages, consider https://pythonwheels.com/
+	$(PYTHON) -m pip install --upgrade pip setuptools
+	# TODO: pinned/unpinned dependency version.
+	# See https://hkn-mu.atlassian.net/browse/COMPSERV-110
+	$(PYTHON) -m pip install -r requirements.txt
 
-.PHONY: venv
-venv: Pipfile Pipfile.lock
-	pipenv install --dev
+.PHONY: install
+install:
+	make install-prod
+	$(PYTHON) -m pip install -r requirements-dev.txt
 
 .PHONY: createsuperuser superuser
 superuser: createsuperuser
 createsuperuser:
-	HKNWEB_MODE='dev' pipenv run python ./manage.py createsuperuser
+	$(MANAGE) createsuperuser
 
 .PHONY: migrate
 migrate:
-	HKNWEB_MODE='dev' pipenv run python ./manage.py migrate
+	$(MANAGE) migrate
 
 .PHONY: makemigrations migrations
 migrations: makemigrations
 makemigrations:
-	HKNWEB_MODE='dev' pipenv run python ./manage.py makemigrations
+	$(MANAGE) makemigrations
 
 .PHONY: shell
 shell:
-	HKNWEB_MODE='dev' pipenv run python ./manage.py shell
+	$(MANAGE) shell
 
 .PHONY: test
 test: venv
-	pipenv run pytest -v tests/
-	pipenv run pre-commit run --all-files
+	$(BIN)/pytest -v tests/
+	$(BIN)/pre-commit run --all-files
```

Comment:
This may not have the intended effect, you may want to actually source the virtualenv here somehow. Strong suspicion this call will pick up the system Python.

Reply:
Welp, definitely is the case that there's an error `make: execvp: .venv/bin/pytest: Permission denied`. Tried `chmod +x $(BIN)/pytest`, but same error

Reply:
Funnily enough, this is the same error that I got when trying to use Node module executables (not sure how I solved that issue either, but it's in the course surveys frontend branch [here](https://github.com/bri25yu/hknweb/blob/course-surveys-frontend/provision_frontend.sh)

Reply:
I'm surprised, this works fine for me after creating a new virtualenv (`python3 -m venv .venv`) and installing pytest and pre-commit (`pip install pytest pre-commit`):
```
jvperrin@jaws:~$ .venv/bin/pytest --version
pytest 6.1.2
jvperrin@jaws:~$ .venv/bin/pre-commit --version
pre-commit 2.8.2
```

Reply:
Yeah, even on a fresh install of the vm, this still occurs ://. Tried one more time `chmod -R 777 .venv` but still same permission denied error.

Reply:
As this is the only issue, I'm going to just make an issue for this and merge.